#!/usr/bin/env python3
PACKAGE = "spinnaker_camera_driver_ros"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("exposure",    int_t,    0, "Exposure", 0,  0, 20000)
gen.add("gain",    double_t,    0, "Gain", 0,  0, 27)

gen.add("grey_value",    double_t,    0, "Grey value", 0,  0, 90)
gen.add("ev_comp",    double_t,    0, "EV Compensation", 0.0,  -3, 3)

gen.add("gamma",    double_t,    0, "Gamma", 0.6,  0.25, 2)
gen.add("black_level",    double_t,    0, "Black level", 0.0,  0.0, 10.0)
gen.add("balance_ratio",    double_t,    0, "Balance ratio", 0,  0, 4)

gen.add("jpeg_quality",    int_t,    0, "Jpeg quality", 90,  0, 100)

gen.add("max_framerate",    double_t,    0, "Max Framerate", 4,  1.6, 32)

binning_enum = gen.enum([ gen.const("Full",      int_t, 1, "Full resolution (1)"),
                       gen.const("Half",     int_t, 2, "Half resolution (2)")],
                      "Binning sizes")

gen.add("binning",    int_t,    0, "Binning", 1,  1, 2, edit_method=binning_enum)


exit(gen.generate(PACKAGE, "spinnaker_camera_driver_ros", "CameraArray"))
